### PR TITLE
Add Azure DNS

### PR DESCRIPTION
**Pull request by vatsalyagoel**
Adds Azure DNS Provider. Used AWS Route53 as base solves  #42

### CODE REVIEW

**Comment by vatsalyagoel** [comment]
Oops did not recognise @captncraig also submitted a PR. Just finished working on it.

**Comment by captncraig** [comment]
Its actually @mhenderson-so who did the work, but yeah. 

1. Thanks for the contribution. A new provider is a big effort and should be commended.
2. I'm gonna give this a good look over regardless. It may make sense to use this vs the other one. Or transplant ideas to/from the other one.
3. Are you able to commit to maintaining this provider in the future?

**Comment by vatsalyagoel** [comment]
I would commit to maintaining the provider as the need came from using it at the place I work. I went over @mhenderson-so 's PR. And he solved some issues that I had regarding the two tests that failed and retry policies.

I think a combination of both the Pull Requests would make sense.

**Comment by mhenderson-so** [comment]
This is awesome. I have no real attachment to my PR, so feel free to take as much of my stuff as you like. We both solved most of it in the same method (and we both used Route 53 as a base). I'll let @captncraig do what he wants with the merging and then close my PR.

**Comment by tlimoncelli** [comment]
Is there an emoji for "I love cooperation!" ???

**Comment by mhenderson-so** [comment]
👬🤝

**Comment by captncraig** [comment]
Gonna work on reviewing this now. Could you please add the following to `build/azure-pipelines/integration.yml`:

```
- job: Azure
  steps:
  - template: go-env.yaml
  - script: go test -v -verbose -provider AZURE
    workingDirectory: $(wd)
    env:
      AZURE_CLIENT_ID: $(AZ_CLIENTID)
      AZURE_CLIENT_SECRET: $(AZ_CLIENTSECRET)
      AZURE_DOMAIN: $(AZ_DOMAIN)
      AZURE_RESOURCE_GROUP: $(AZ_RSGNAME)
      AZURE_SUBSCRIPTION_ID: $(AZ_SUBSCRIPTIONID)
      AZURE_TENANT_ID: $(AZ_TENANTID)
```

so that I can run the main integration suite with your code.

File: providers/azuredns/azureDnsProvider.go [hunk 1]
```diff
@@ -0,0 +1,430 @@
...
+
+	return nil
+}
+
+type errNoExist struct {
+	domain string
+}
+
+func (e errNoExist) Error() string {
+	return fmt.Sprintf("Domain %s not found in your route Azure account", e.domain)
```
Comment (captncraig) [inline providers/azuredns/azureDnsProvider.go]: remove "route"

File: integrationTest/providers.json [hunk 2]
```diff
@@ -4,6 +4,15 @@
...
     "knownFailures": "29,30,31,32,33,34,35,38,39,40,41,48,49,51,52,53"
   },
+  "AZURE_DNS": {
+    "SubscriptionID": "$AZURE_SUBSCRIPTION_ID",
+    "ResourceGroup": "$AZURE_RESOURCE_GROUP",
+    "TenantID": "$AZURE_TENANT_ID",
+    "ClientID": "$AZURE_CLIENT_ID",
+    "ClientSecret": "$AZURE_CLIENT_SECRET",
+    "domain": "$AZURE_DOMAIN",
+    "knownFailures": "16,26"
```
Comment (captncraig) [inline integrationTest/providers.json]: What are these failures? I hope we can make them pass with some creativity.

**Comment by vatsalyagoel** [comment]
@captncraig Cleaned up code and fixed the broken tests. Also added the integration test to azure-pipelines.yml file

**Comment by captncraig** [comment]
🎱

**Comment by captncraig** [comment]
Thank you for this. Looks great!

**Comment by tlimoncelli** [comment]
@vatsalyagoel IS AWESOME!!!

**Pull request merged**